### PR TITLE
Add UCPD peripheral

### PR DESCRIPTION
**Pull request by RootCubed**
Adds the UCPD peripheral available in G0x1, G4, H7R/S, H5, L5 and U5 (see https://www.st.com/content/st_com/en/ecosystems/stm32-usb-c.html, "Key products with built-in UCPD controller").

### CODE REVIEW

**Comment by burrbull** [comment]
Looks good to me.
Although for new patches I prefer to place `_include`s under peripheral in root yaml:
```yaml
PER:
  _include:
    - patch_without_per_name.yaml
```
With no peripheral name in inluded files. This is slightly more verbose, but avoids such `UCPD,UCPD?:` hacks and makes patch execution order more predictable.

**Comment by RootCubed** [comment]
Makes sense, I'll change that tomorrow.

**Review by burrbull — APPROVED** [review]

File: peripherals/ucpd/ucpd_v1.yaml [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    Enabled: [1, "SOP'_Debug enabled"]
+  RXORDSETEN6:
+    Disabled: [0, "SOP''_Debug disabled"]
+    Enabled: [1, "SOP''_Debug enabled"]
+  RXORDSETEN7:
+    Disabled: [0, "SOP extension #1 disabled"]
+    Enabled: [1, "SOP extension #1 enabled"]
+  RXORDSETEN8:
+    Disabled: [0, "SOP extension #2 disabled"]
+    Enabled: [1, "SOP extension #2 enabled"]
```
Comment (burrbull) [inline peripherals/ucpd/ucpd_v1.yaml:40]: Only one note.
I would prefer description to be in field description.
With 1 common enum.
```suggestion
  # Move this part to another file in common_patches
  _delete: RXORDSETEN
  _add:
    RXORDSETEN0:
      description: SOP detection
      bitOffset: 20
      bitWidth: 1
    RXORDSETEN1:
      description: SOP' detection
      bitOffset: 21
      bitWidth: 1
    RXORDSETEN2:
      description: SOP'' detection
      bitOffset: 22
      bitWidth: 1
    RXORDSETEN3:
      description: Hard Reset detection
      bitOffset: 23
      bitWidth: 1
    RXORDSETEN4:
      description: Cable Detect reset
      bitOffset: 24
      bitWidth: 1
    RXORDSETEN5:
      description: SOP'_Debug
      bitOffset: 25
      bitWidth: 1
    RXORDSETEN6:
      description: SOP'' Debug
      bitOffset: 26
      bitWidth: 1
    RXORDSETEN7:
      description: "SOP extension #1" 
      bitOffset: 27
      bitWidth: 1
    RXORDSETEN8:
      description: "SOP extension #1"
      bitOffset: 28
      bitWidth: 1
  
  # Leave this here
  RXORDSETEN?:
    Disabled: [0, Flag disabled]
    Enabled: [1, Flag enabled]
```

Comment (RootCubed) [reply peripherals/ucpd/ucpd_v1.yaml:40]: From where should I include that RXORDSETEN patch? In every `stm32xxx.yaml` that has UCPD or can I just put it once in `ucpd_v1.yaml`?

Comment (burrbull) [reply peripherals/ucpd/ucpd_v1.yaml:40]: Ideally `peripheral` folder should contain only addition of enumeratedValues and writeConstains.
This is why I've asked to move it.
Although I does not insist.

**Review by burrbull — APPROVED** [review]

**Pull request merged**